### PR TITLE
circleci: allow approval before build

### PR DESCRIPTION
**Pull request by kopiczko**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -151,85 +151,70 @@ workflows:
...
-          filters:
-            branches:
-              only: master
           requires:
-          - build
+          - master
       - e2eTestCurPRUpdate:
           requires:
-          - hold
+          - pr
```
Comment (kopiczko) [inline .circleci/config.yml]: This is

```yaml
workflows:
  version: 2
  build_e2e:
    jobs:
      - build
      - hold:
          type: approval
      - master:
          requires:
            - build
          filters:
            branches:
              only: master
      - pr:
          requires:
            - build
            - hold
          filters:
            branches:
              ignore: master



      - e2eTestCurMasterClusterState:
          requires:
            - master
      - e2eTestWIPMasterClusterState:
          requires:
            - master
      - e2eTestCurPRClusterState:
          requires:
            - pr
      - e2eTestWIPPRClusterState:
          requires:
            - pr



      - e2eTestCurMasterDraining:
          requires:
            - master
      - e2eTestWIPMasterDraining:
          requires:
            - master
      - e2eTestCurPRDraining:
          requires:
            - pr
      - e2eTestWIPPRDraining:
          requires:
            - pr



      - e2eTestCurMasterScaling:
          requires:
            - master
      - e2eTestWIPMasterScaling:
          requires:
            - master
      - e2eTestCurPRScaling:
          requires:
            - pr
      - e2eTestWIPPRScaling:
          requires:
            - pr



      - e2eTestCurMasterUpdate:
          requires:
          - master
      - e2eTestCurPRUpdate:
          requires:
          - pr
```

Comment (kopiczko) [reply .circleci/config.yml]: screen shot 2018-10-25 at 12 50 38

🎉

File: .circleci/config.yml [hunk 2]
```diff
@@ -151,85 +151,70 @@ workflows:
...
+      - e2eTestCurPRScaling:
+          requires:
+            - pr
+      - e2eTestWIPPRScaling:
+          requires:
+            - pr
 
 
 
       - e2eTestCurMasterUpdate:
```
Comment (kopiczko) [inline .circleci/config.yml]: @xh3b4sd should we run these in WIP as well?

File: .circleci/config.yml (hunk 1 above)
Comment (MarcelMue) [reply .circleci/config.yml]: You will need to define `pr` or `master` jobs here as well: https://github.com/giantswarm/aws-operator/pull/1221/files#diff-1d37e48f9ceff6d8030570cd36286a61R25

File: .circleci/config.yml (hunk 2 above)
Comment (xh3b4sd) [reply .circleci/config.yml]: There should be a comment on the other definition that this makes no sense to be run in WIP. The test takes the current and wip version anyway in order to perform the upgrade. That is why the upgrade test is only there once and not twice.

Comment (kopiczko) [reply .circleci/config.yml]: To clarify. "There should be a comment" means the comment should be there but it disappeared or it should be added?

**Comment by kopiczko** [comment]
I'm going with that.

File: .circleci/config.yml (hunk 1 above)
Comment (kopiczko) [reply .circleci/config.yml]: Works also for master. As you can see hold is not green and tests are running.

screen shot 2018-10-25 at 17 30 07

**Pull request merged**